wikichange/src/content/content.js

```diff
@@ -79,31 +79,86 @@ const renderGraphOverlay = async () => {
             clearTimeout(sim);
         }
         percentage++;
-    }
+    };
     let siteSub = document.getElementById("siteSub");
     insertAfter(floatContainer, siteSub);
     const creationDate = await getPageCreationDate(title);
     const sim = setInterval(progressBar, 3);
-    injectGraphToPage(title, creationDate, new Date(Date.now()));
+    injectGraphToPage(title, creationDate, new Date(Date.now())).then(() => document.getElementById("5y").click());
+
+    renderScaleButtons();
+};
+
+const setUpScaleButton = (scaleButtonsDiv, buttonId, buttonText, duration, scaleButtonInputs) => {
+    const button = document.createElement("button");
+    button.setAttribute("id", buttonId);
+    button.setAttribute("style", "margin-right: 5px;");
+    button.innerHTML = buttonText;
+    scaleButtonsDiv.appendChild(button);
+
+    button.addEventListener("click", () => {
+        injectScaledCurrentGraphToPage(duration);
+
+        // remove hover effect from all scale buttons
+        scaleButtonInputs.forEach((input) => {
+            document.getElementById(input.id).className = "buttonNoHoverEffect";
+        });
+
+        button.className = "buttonHoverEffect";
+    });
+};
+
+const setUpScaleButtons = (scaleButtonsDiv, scaleButtonInputs) => {
+    scaleButtonInputs.forEach((input) => {
+        setUpScaleButton(scaleButtonsDiv, input.id, input.name, input.duration, scaleButtonInputs);
+    });
+};
+
+const getDateObjectFromNow = (monthsFromCurrent) => {
+    const date = new Date();
+    date.setMonth(date.getMonth() - monthsFromCurrent);
+    return date;
+};
+
+/**
+ * Render buttons to scale the graph.
+ */
+const renderScaleButtons = () => {
+    const viewsEditsChart = document.getElementById("viewsEditsChart");
+    const scaleButtonsDiv = document.createElement("div");
+    scaleButtonsDiv.setAttribute("id", "scaleButtonsDiv");
+    scaleButtonsDiv.setAttribute("style", "text-align: start;");
+
+    const scaleButtonInputs = [
+        { id: "all", name: "ALL", duration: null }, // null represents shows everything
```

Comment:
Need some more context here on the button's names. or some text saying this is time period for the graph - Sneer told us we need more context

Reply:
oh yup forgot about that. will add sth

Reply:
On second thought, that comment was before we have the highlight thingy on the current highlight. it might not be needed anymore (adding more text doesn't mean that people will read it lol). Will keep it like this for now.